src/lib/rest/StringFilter.h

```diff
@@ -0,0 +1,138 @@
+*/
+#include <string>
+#include <vector>
+
+#include "mongo/client/dbclient.h"
+
+#include "ngsi/ContextAttribute.h"
+
+using namespace mongo;
+
+
+
+/* ****************************************************************************
+*
+* StringFilterOp - 
+*/
+typedef enum StringFilterOp
+{
+  SfopExists,              //
+  SfopNotExists,           // !
+  SfopEquals,              // ==
+  SfopDiffers,             // !=
+  SfopGreaterThan,         // >
+  SfopGreaterThanOrEqual,  // >=
+  SfopLessThan,            // <
+  SfopLessThanOrEqual      // <=
+} StringFilterOp;
+
+
+
+/* ****************************************************************************
+*
+* StringFilterValueType - 
+*/
+typedef enum StringFilterValueType
+{
+  SfvtString,
+  SfvtBool,
+  SfvtNumber,
+  SfvtNull,
+  SfvtDate,
+  SfvtNumberRange,
+  SfvtDateRange,
+  SfvtStringRange,
+  SfvtNumberList,
+  SfvtDateList,
+  SfvtStringList
+} StringFilterValueType;
+
+
+
+/* ****************************************************************************
+*
+* StringFilterItem - 
+*/
+class StringFilterItem
+{
+public:
+  std::string               left;
+  StringFilterOp            op;
+  StringFilterValueType     valueType;
+  double                    numberValue;
+  std::string               stringValue;
+  bool                      boolValue;
+  std::vector<std::string>  stringList;
+  std::vector<double>       numberList;
+  double                    numberRangeFrom;
+  double                    numberRangeTo;
+  std::string               stringRangeFrom;
+  std::string               stringRangeTo;
+  std::string               attributeName;  // Used for unary operators only
+
+  bool                      parse(char* qItem, std::string* errorStringP);
+  const char*               opName(void);
+  const char*               valueTypeName(void);
+
+  bool                      valueGet(char*                   s,
+                                     StringFilterValueType*  valueTypeP,
+                                     double*                 doubleP,
+                                     std::string*            stringP,
+                                     bool*                   boolP,
+                                     std::string* errorStringP);
+
+  bool                      valueParse(char* s, std::string* errorStringP);
+  bool                      rangeParse(char* s, std::string* errorStringP);
+  bool                      listParse(char* s, std::string* errorStringP);
+  bool                      listItemAdd(char* s, std::string* errorStringP);
+  bool                      matchEquals(ContextAttribute* caP);
+  bool                      matchGreaterThan(ContextAttribute* caP);
+  bool                      matchLessThan(ContextAttribute* caP);
+};
+
+
+
+class ContextElementResponse;
+/* ****************************************************************************
+*
+* StringFilter - 
+*/
+class StringFilter
+{
+public:
+  std::vector<StringFilterItem>  filters;
```

Comment:
Some comments explaining this filed should be added. For example, what is plainString? The string representing the q filter? Regarding mongoFilters, it should be clarified that this field needs the call to mongoFilterPopulated() to be used. Do we need to call parse() before calling mongoFilterPopulate() and/or match()?

I thing adding some hint covering these topics will help if we later need to use or mantain this stuff.